book/AIDO/31_task_embodied_strategies/38_il_sim_dagger.md

```diff
@@ -1,45 +1,91 @@
 # Dataset Aggregation {#embodied_il_sim_dagger status=ready}
 
-This section describes the procedure for training and testing an agent on [gym-duckietown](https://github.com/duckietown/gym-duckietown) using the [Dagger](https://www.cs.cmu.edu/~sross1/publications/Ross-AIStats11-NoRegret.pdf) algorithm.
-It can be used as a starting point for any of the [`LF`](#lf), [`LFV`](#lf_v), and [`LFVI`](#lf_v_i) challenges.
+This section describes the procedure for training and testing an agent with the [gym-duckietown](https://github.com/duckietown/gym-duckietown) simulator using the [Dagger](https://www.cs.cmu.edu/~sross1/publications/Ross-AIStats11-NoRegret.pdf) algorithm.
+
+It can be used as a starting point for any of the [`LF`](#challenge-LF), [`LFP`](#challenge-LFP), and [`LFV_multi`](#challenge-LFV_multi) challenges.
 
 
 <div class='requirements' markdown='1'>
 
-Requires: You already know something about PyTorch.
+Requires: You are somewhat familiar with PyTorch and the [Pytorch template](#pytorch-template).
 
 Result: You could win the AI-DO!
 
 </div>
 
+## Introduction
+
+We saw a first implementation of immitation learning in the behaviour cloning baseline. That baseline models the driving task as an end-to-end supervised learning problem where data can be collected offline from an expert. One of the central issues with this approach is that of **distributional shift**. Since this is a sequential decision making problem, the training data are not "identically and independently distributed". The result is that if your agent deviates from the *optimal* trajectory that was demonstrated by the expert, it will not have any data in its dataset that shows it how to *recover back* to the optimal trajectory. As a result, it is unlikely that the behiaviour cloning approach will be robust.  
+
+For a better result than behaviour cloning this second version of imitation learning does not train only on a single trajectory given by the expert. We follow the Dataset Aggreagation algorithm [(Dagger)](https://www.cs.cmu.edu/~sross1/publications/Ross-AIStats11-NoRegret.pdf) where we also let the agent interact with the environment and allow the expert to *recover*.  The actions between the expert and the learner are chosen randomly with a varying probability with the hope that the expert _corrects_ the learner if it starts deviating from the optimal trajectory.
+
 ## Quickstart 
 
-1) Clone this [repo](https://github.com/duckietown/gym-duckietown):
```

Comment:
We should mention installing gym duckietown before cloning and running the IL baseline